core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package job
+
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import cats.effect.{Concurrent, Timer}
+
+import fs2.concurrent.{Queue, SignallingRef}
+
+import scala.{Array, Boolean, Int, List, Option, None, Nothing, Some, Unit}
+import scala.collection.JavaConverters._
+import scala.util.{Left, Right}
+
+import java.lang.{RuntimeException, SuppressWarnings}
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * A coordination mechanism for parallel job management. This
+ * structure provides mechanisms for aggregated notifications
+ * from jobs, deterministic cancelation, "fire and forget"
+ * submission, and sidecar-style cancelation of direct streams.
+ */
+final class JobManager[F[_]: Concurrent: Timer, I, N] private (
+    notificationsQ: Queue[F, Option[N]],
+    dispatchQ: Queue[F, Stream[F, Nothing]]) {
+
+  import JobManager._
+
+  private[this] val meta: ConcurrentHashMap[I, Context[F]] = new ConcurrentHashMap[I, Context[F]]
+
+  val notifications: Stream[F, N] = notificationsQ.dequeue.unNoneTerminate
```

Comment:
Might be useful to include the id in the stream, so consumers can identify the job.

Reply:
Or maybe something like `I => Stream[F, N]` which may also solve the backpressure issue since notifications would only apply backpressure if there is a consumer.